pkg/agent/utils.go

```diff
@@ -194,6 +195,40 @@ func getBase64EncodedGzippedCustomScript(csFilename string, config *datamodel.No
 	return getBase64EncodedGzippedCustomScriptFromStr(csStr)
 }
 
+func removeComments(b []byte) []byte {
+	var contentWithoutComments []string
+	lines := strings.Split(string(b), "\n")
+	for _, line := range lines {
+		lineNoWhitespace := strings.TrimSpace(line)
+		if isCommentInBeginningOfLine(lineNoWhitespace) {
+			// ignore entire line that is a comment
+			continue
+		}
+		lastHashIndex := strings.LastIndex(line, "#")
+		if lastHashIndex > 0 && isCommentAtTheEndOfLine(lastHashIndex, line) {
+			// remove only the comment part from line
+			line = line[:lastHashIndex]
+		}
+		contentWithoutComments = append(contentWithoutComments, line)
+	}
+	return []byte(strings.Join(contentWithoutComments, "\n"))
+}
+
+// Trying to avoid using a regex. There are certain patterns we ignore just to be on the safe side. This is enough to get rid of most of the obvious comments.
+func isCommentAtTheEndOfLine(lastHashIndex int, trimmedToCheck string) bool {
+	getSlice := func(start, end int, str string) string {
+		if end > len(str) || start > end {
```

Comment:
does the index-range operator not already take care of this?

Reply:
This function just checks if specific patterns appear in the line, to access it "safely" i added this getSlice function. So I cannot easily replicate this with for-range. Or did you mean something else?

Reply:
oh, yeah I guess this operator isn't smart enough, just tested locally